examples/advanced/sklearn-linear/job.py

```diff
@@ -0,0 +1,142 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This code shows how to use NVIDIA FLARE Job Recipe to run federated learning with scikit-learn
+linear models using FedAvg algorithm.
+
+Per-Client Data Splits:
+    The old prepare_job_config.sh approach is replaced by passing a dict for train_args.
+    This allows different data ranges for each client, supporting non-IID scenarios.
+"""
+
+import argparse
+
+from nvflare.app_opt.sklearn import SklearnFedAvgRecipe
+from nvflare.recipe import SimEnv
+
+
+def define_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--n_clients", type=int, default=5, help="Number of clients")
+    parser.add_argument("--num_rounds", type=int, default=50, help="Number of training rounds")
+    parser.add_argument(
+        "--data_path",
+        type=str,
+        default="/tmp/nvflare/dataset/HIGGS.csv",
+        help="Path to HIGGS dataset CSV file",
+    )
+    parser.add_argument(
+        "--split_method",
+        type=str,
+        default="uniform",
+        choices=["uniform", "custom"],
+        help="Data split method: 'uniform' (same args for all) or 'custom' (per-client ranges)",
+    )
+
+    return parser.parse_args()
+
+
+def calculate_data_splits(n_clients: int, total_size: int = 11000000, valid_size: int = 1100000):
+    """Calculate uniform data splits for clients.
```

Comment:
is our "custom" and "uniform" right now actually the same strategy? both mean "uniform"

in that case should we just dont say split method,
and ask Users to implement this calculate_data_splits if they need custom